pkg/pillar/types/zedroutertypes.go

```diff
@@ -2889,3 +2921,116 @@ func (status OnboardingStatus) LogDelete(logBase *base.LogObject) {
 func (status OnboardingStatus) LogKey() string {
 	return string(base.OnboardingStatusLogType) + "-" + status.Key()
 }
+
+// Bitmap :
+// Bitmap of the reserved and allocated resources
+// Keeps 256 bits indexed by 0 to 255.
+type Bitmap [32]byte
+
+// IsSet :
+// Test the bit value
+func (bits *Bitmap) IsSet(i int) bool {
+	return bits[i/8]&(1<<uint(7-i%8)) != 0
+}
+
+// Set :
+// Set the bit value
+func (bits *Bitmap) Set(i int) {
+	bits[i/8] |= 1 << uint(7-i%8)
+}
+
+// Clear :
+// Clear the bit value
+func (bits *Bitmap) Clear(i int) {
+	bits[i/8] &^= 1 << uint(7-i%8)
+}
+
+// AddToIP :
+func AddToIP(ip net.IP, addition int) net.IP {
+	if addr := ip.To4(); addr != nil {
+		val := uint32(addr[0])<<24 + uint32(addr[1])<<16 +
+			uint32(addr[2])<<8 + uint32(addr[3])
+		val += uint32(addition)
+		byte0 := byte((val >> 24) & 0xFF)
+		byte1 := byte((val >> 16) & 0xFF)
+		byte2 := byte((val >> 8) & 0xFF)
+		byte3 := byte(val & 0xFF)
+		return net.IPv4(byte0, byte1, byte2, byte3)
+	}
+	//TBD:XXX, have function for IPv6
+	return net.IP{}
+}
+
+// GetIPSubnetBase :
+func GetIPSubnetBase(ip, gateway net.IP) int {
+	if addr := ip.To4(); addr != nil {
+		val0 := uint32(addr[0])<<24 + uint32(addr[1])<<16 +
+			uint32(addr[2])<<8 + uint32(addr[3])
+		if gateway == nil {
+			return 0
+		}
+		addr = gateway.To4()
+		if addr == nil {
+			return 0
+		}
+		val1 := uint32(addr[0])<<24 + uint32(addr[1])<<16 +
+			uint32(addr[2])<<8 + uint32(addr[3])
+		return int(val1 - val0)
+	}
+	//TBD:XXX, have function for IPv6
+	return 0
+}
+
+// GetIPAddrCountOnSubnet IP address count on subnet
+func GetIPAddrCountOnSubnet(subnet net.IPNet) int {
+	prefixLen, _ := subnet.Mask.Size()
+	if net.ParseIP(subnet.IP.String()).To4() != nil {
+		return 0x01 << (32 - prefixLen)
+	}
+	if net.ParseIP(subnet.IP.String()).To16() != nil {
+		return 0x01 << (128 - prefixLen)
+	}
+	return 0
+}
+
+// GetHostCountOnSubnet addressable host count on subnet
+func GetHostCountOnSubnet(subnet net.IPNet) int {
+	if addrCount := GetIPAddrCountOnSubnet(subnet); addrCount != 0 {
+		// discount the gateway, network and broadcast addresses(3).
+		addrCount -= 3
+		return getMaxHostCountOnUNet(addrCount)
+	}
+	return 0
+}
+
+// AppNumber :
+// PS. Any change to AppNumMax, must be
+// reflected in the BitMap Size(8 bytes)
+// getMaxHostCountOnSubnet() and
+// getMaxHostCountOnDhcpRange() should also be reviewed
+const (
+	AppNumStart   = 0   // DhcpRange start
+	AppNumEnd     = 252 // DhcpRange end
+	AppNumMax     = 253 // Max Count
+	LargeAppCount = 16  // to determine the default Dhcp Range
+	AppNumInvalid = 255
+)
+
+// getMaxHostCountOnUNet restricted by the Bitmap Size
+// for network size more than types.AppNumMax(253), restrict it.
+// limited by bitmap size, with a discount of 3
+// (network, broadcast and gateway address)
```

Comment:
This doesn't subtract 3 hence I wonder where that is supposed to happen.

Reply:
For instance, if I have a /26 subnet, it should only fit 64-3.

Reply:
Ah the problem is that getMaxHostCountOnUNet does not make sense as a function definition and this comment and logic all belongs in the caller which does the subtraction of 3. So please fold this into the caller.